### ISSUE TITLE
Shadows for partially-transparent Meshes

### ISSUE THREAD

**Issue by Bug-Reaper** [Ameobea/three-good-godrays#11]
I have a sparsely-transparent skybox I want to use over a point-light as a shader-driven lamp-shade of sorts.

<img width="939" alt="Screenshot 2024-12-30 at 8 06 41 PM" src="https://github.com/user-attachments/assets/db697564-3b8f-4c58-89d4-f1ba75000179" />

I don't know if support for semi-transparent objects is possible since depth is still the same?

**Comment by Ameobea** [comment]
Yeah, transparent objects won't work very well with the godrays because the shader uses the depth buffer and there's no information about transparency available there.

I'm not sure exactly what you're trying to achieve with your user case, but if you just need to ignore the transparent objects you might be able to set up a multi-pass render pipeline with the pmndrs `postprocessing` library and render transparent objects separately from the others before or after the godrays pass.

**Comment by Bug-Reaper** [comment]
I was trying to use a sphere with mixed alpha texture as point-light lamp-shade of sorts. So flat out ignoring is no good. 

I may at some point tinker with adding some-type of alpha support. Currently too much on my plate.
